### ISSUE TITLE
Cannot UPDATE `pg_aoseg` tables

### ISSUE THREAD

**Issue by d** [greenplum-db/gpdb#7355]
### Greenplum version or build
Master (7.0.0 alpha 0 as of writing) and 6X_STABLE.

### Step to reproduce the behavior

```sql
set allow_system_table_mods = on;

UPDATE pg_aoseg.pg_aoseg_17431 SET eof=184;
```

The plan is busted: the lower scan node doesn't correctly project the ctid.

Collaborators are welcome to fill in more details. Will come back later.

**Comment by gfphoenix78** [comment]
The relkind of pg_aoseg.pg_aoseg_<oid> is RELKIND_AOSEGMENTS, so the ctid is not assigned as normal relations.
And the generated planner is different with ordinary table.
  see **`EXPLAIN UPDATE pg_aoseg.pg_aoseg_17431 SET eof=184;`**

**Comment by ashwinstar** [comment]
We should fix this, it's helpful for fixing sometimes problematic pg_aoseg's. One of the trick/workaround we recently had used was to first mark in pg_class pg_aoseg table's relkind as 'r', update it and then mark it back as RELKIND_AOSEGMENTS. Having this issue fixed will help to avoid such a dance.

**Comment by dpandhi-git** [comment]
with @ashwinstar fixed in 7x 232d1565c3c915a663f0621beb1f9691f1e8cc3f
